src/cheat_gui.cpp

```diff
@@ -56,7 +56,7 @@ static int32 _money_cheat_amount = 10000000;
  */
 static int32 ClickMoneyCheat(int32 p1, int32 p2)
 {
-	Command<CMD_MONEY_CHEAT>::Post(p2 * _money_cheat_amount);
+	Command<CMD_MONEY_CHEAT>::Post(Money(p2) * _money_cheat_amount);
```

Comment:
Conceptually I would do `Money(_money_cheat_amount) * p2` as the amount is the actual money, whereas p2 is just a multiplier.